### ISSUE TITLE
MSI B450 Mortar pulses when not told to.

### ISSUE THREAD

**Issue by westjgames** [nagisa/msi-rgb#73]
`sudo ./msi-rgb FFFFFFFF 22220000 DDDDDDDD -ir -ib -ig`

This results in the beautiful purple colour I wanted, but it's always pulsing, despite me not adding that flag.

Adding `--pulse` causes the lights to not turn on at all.

I get the feeling this board expects the inverse of everything it's sent? (I had to invert all the colour values to get the one I wanted)

**Comment by rockybulwinkle** [comment]
I'm experiencing the same issue on the B450 tomahawk with a Ryzen 3600. I get a purple pulsing color with westjgames' command.

**Comment by rockybulwinkle** [comment]
It also turns off for me when I pass the --pulse argument.

**Comment by AzaIndustries** [comment]
Yeah I've tried all kinds of combinations to try and remove the pulsing.
Either it pulses with all inverted:
msi-rgb ff0000ff 0000ff00 00ff00ff  -ir -ig  -ib 
Or it's completely off with inverted+pulse:
msi-rgb ff0000ff 0000ff00 00ff00ff  -ir -ig  -ib --pulse
Or it pulses just blue/green/white with:
msi-rgb ff0000ff 0000ff00 00ff00ff 

Colour values are a bit weird but I can deal with that.
I just want solid color transitions. =/

**Comment by rockybulwinkle** [comment]
> Yeah I've tried all kinds of combinations to try and remove the pulsing.
> Either it pulses with all inverted:
> msi-rgb ff0000ff 0000ff00 00ff00ff -ir -ig -ib
> Or it's completely off with inverted+pulse:
> msi-rgb ff0000ff 0000ff00 00ff00ff -ir -ig -ib --pulse
> Or it pulses just blue/green/white with:
> msi-rgb ff0000ff 0000ff00 00ff00ff
> 
> Colour values are a bit weird but I can deal with that.
> I just want solid color transitions. =/

Yeah.... I think someone needs to look into how the MSI's b450 boards work a little more. I may take a look myself if I find myself with the time.

**Comment by westjgames** [comment]
#79

**Comment by AzaIndustries** [comment]
Is it just inverted? It's really confusing me while trying to work it out.
Maybe I'm not understanding how the software works, but I can't make head or tails of any discernible pattern no matter what I try to deduce.
Especially when there's all kinds of bizarre fading and non-responsive settings.

**Comment by xorrito** [comment]
I can attest that I too have this issue. I have the B450 Tomahawk with the ryzen 7 2700 LED cooler. the cooler connected to the LED header follows the same issue as the on board LEDs. If there is anything I can do to help, let me know.

**Comment by bpiero** [comment]
#102

**Comment by pradyunsg** [comment]
For anyone wondering what's relevant about @bpiero's referenced issue, the first sentences are:

> My MB (mortar max) was sometimes turning mad and pulsing when not told to, and I noticed that it can't be stopped ( sudo ./target/release/msi-rgb FFFFFFFF FFFFFFFF FFFFFFFF --disable ). I suspected that there should be some kind of memory that wasn't flushed, so I did reboot under windows and I turned off the lights in Mystic Light, reboot under linux and it solved the issue.

Based on this, one solution for this problem seems to be:

- Boot into Windows
- Use MSI Mystic Light to disable all LEDs
- Boot into Linux
- No more pulsing!
